group/dleq/dleq.go

```diff
@@ -0,0 +1,260 @@
+// Package dleq allows create proofs of Discrete-Logarithm Equivalence (DLEQ).
+package dleq
+
+import (
+	"crypto"
+	"encoding/binary"
+	"errors"
+	"io"
+
+	"github.com/cloudflare/circl/group"
+)
+
+const (
+	labelSeed         = "Seed-"
+	labelChallenge    = "Challenge"
+	labelComposite    = "Composite"
+	labelHashToScalar = "HashToScalar-"
+)
+
+type Params struct {
```

Comment:
Maybe call this type `Prover`, and the corresponding function `Prove`? That way, the usage would look like:

```
prover := dleq.Prover{...}
proof, err := prover.Prove(k, A, kA, b, kB, rnd)
```

Reply:
done